### PR TITLE
Consistent javadoc formatting

### PR DESCRIPTION
**Pull request by radcortez**
There are a lot of changes here, but I didn't modify the content itself.

This work was mostly to make the docs consistent between all sources:

- Removed duplicated empty lines.
- Removed double spaces.
- Wrap at 120 chars.
- Same indentations for li and ol.
- Use paragraphs where required.
- Fixed javadoc warnings.

### CODE REVIEW

**Comment by Emily-Jiang** [comment]
@radcortez I think it might be better to change the formatting rules. If we use different rules, we will be run into constant battle of changing.

File: api/src/main/java/org/eclipse/microprofile/config/inject/package-info.java [hunk 1]
```diff
@@ -16,35 +15,35 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- *******************************************************************************/
+ */
 
 /**
- * <p>CDI Support for Microprofile Config
+ * CDI Support for Microprofile Config
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/inject/package-info.java:21]: MicroProfile

File: api/src/main/java/org/eclipse/microprofile/config/inject/package-info.java [hunk 2]
```diff
@@ -16,35 +15,35 @@
...
+ *
+ * <p>
+ * Microprofile Config also supports injection via a JSR-330 DI container:
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/inject/package-info.java:24]: MicroProfile

**Comment by radcortez** [comment]
> @radcortez I think it might be better to change the formatting rules. If we use different rules, we will be run into constant battle of changing.

Yes, I did suggest this previously here:
https://github.com/eclipse/microprofile-config/issues/536

Actually let me do a PR with that and how the code would look like with the changes so we can discuss.

**Pull request merged**